app/src/js/controllers/ImageController.js

```diff
@@ -47,6 +48,44 @@ angular.module('media_manager')
       CourseCache.current_image = CourseCache.images[ic.index];
       resetBreadcrumb();
     }
-  }
+  };
+
+  ic.newLabel = '';
+  ic.newValue = '';
+  ic.saveMetadata = function(label, value){
+    if(label !== undefined && value !== undefined){
```

Comment:
Do you think it's a bad idea to have one save method handle inserts and updates? It seemed like a good idea to me at the time, but for it to be a good idea, I think I need it to actually reuse pieces in each save...